proto/substrait/algebra.proto

```diff
@@ -304,6 +339,7 @@ message RelRoot {
   repeated string names = 2;
 }
 
+// A relation without output field names (used internally in a plan)
```

Comment:
The "without output field names" is strange. Let's state what it is, not what it isn't.

Reply:
I agree it is odd to the specify what is not, but that is a bit suprising/unique for me, so something I felt like calling out. Happy to go either way if you feel strongly.

Reply:
ACK